### DISCUSSION TITLE
Downloading more than 8 demos

### DISCUSSION THREAD

**Discussion by RyanBlakeIT** [akiver/cs-demo-manager#734] [Q&A] [answered]
I see that there's only a maximum of 8 demos you can download at a time.  There are times where I'll play multiple games and then use CSDM to download them, but if I played more than 8, I feel like I'm missing some.  Is there a way to increase the number of demos available to download to a larger number?

**Comment by ClementCariou** [comment]
As far as I know, it's a limitation of CS.

**Comment by akiver** [comment, marked as answer]
The 8 limit is because the Steam API gives only the last 8 games, as mentioned [here](https://cs-demo-manager.com/docs/faq/download#i-can-only-see-my-last-8-games).

> There are times where I'll play multiple games and then use CSDM to download them, but if I played more than 8, I feel like I'm missing some

CS:DM should automatically download the last MM demos if the option "Background download" is enabled:

![dl](https://github.com/akiver/cs-demo-manager/assets/1842524/892f691a-dfaa-4531-85f4-cab816589b4d)

It works only if the CS:DM icon is visible in the system tray (the background process is running).
Demos are downloaded once CS2 is closed, so if you play 10 games without closing CS2 there will be 2 missing games.

**Reply by RyanBlakeIT** [reply]
I run CSDM from my laptop, which I don't use to play CS, but I have it installed.  I do have the CS:DM in my system tray, but it does not automatically download demos, even with that toggle enabled.

**Reply by akiver** [reply]
Demos are downloaded at app startup and when the CS process on your local machine exits.

It can be risky to download demos at a given interval because if you are playing on your PC and your laptop tries to download your last demos, it will disconnect you from the CSGO Steam network (you will be disconnected if you are in-game).

You should either restart the app or refresh the download list from your laptop after closing CS2 on your PC.

**Reply by ClementCariou** [reply]
Personally, I launch CS2 via a script which the waits until the game thread is closed, then it runs csdm CLI to download and analyse the demos. It also handles game restart.

I think csdm could implement directly a similar behavior.

**Reply by akiver** [reply]
CS:DM already does it - not through the CLI, but it's the same.
You only have to make sure the background process is running and the "Background download" option is enabled.
